src/overlays/actors/ovl_En_Talk_Gibud/z_en_talk_gibud.h

```diff
@@ -3,15 +3,108 @@
+    /*  9 */ EN_TALK_GIBUD_ANIMATION_IDLE,
+    /* 10 */ EN_TALK_GIBUD_ANIMATION_WALK,
+    /* 11 */ EN_TALK_GIBUD_ANIMATION_DANCE_SQUAT,
+    /* 12 */ EN_TALK_GIBUD_ANIMATION_DANCE_PIROUETTE,
+    /* 13 */ EN_TALK_GIBUD_ANIMATION_DANCE_CLAP,
+} EnTalkGibudAnimations;
+
+typedef enum {
+    /* 0 */ EN_TALK_GIBUD_TYPE_GIBDO,
+    /* 1 */ EN_TALK_GIBUD_TYPE_REDEAD,
+} EnTalkGibudType;
+
+typedef enum {
+    /* 0 */ EN_TALK_GIBUD_GRAB_START,
+    /* 1 */ EN_TALK_GIBUD_GRAB_ATTACK,
+    /* 2 */ EN_TALK_GIBUD_GRAB_RELEASE,
+} EnTalkGibudGrabState;
+
+typedef enum {
+    /* 0 */ EN_TALK_GIBUD_REQUESTED_ITEM_MET,
+    /* 1 */ EN_TALK_GIBUD_REQUESTED_ITEM_NOT_ENOUGH_AMMO,
+    /* 2 */ EN_TALK_GIBUD_REQUESTED_ITEM_NOT_MET,
+} EnTalkGibudRequestedItemState;
+
+typedef enum {
+    /*  0 */ EN_TALK_GIBUD_LIMB_ROOT,       // Root of Left Leg Root, Right Leg Root, Upper Body Root, and Pelvis
+    /*  1 */ EN_TALK_GIBUD_LIMB_LEFT_LEG_ROOT,
+    /*  2 */ EN_TALK_GIBUD_LIMB_LEFT_THIGH,
+    /*  3 */ EN_TALK_GIBUD_LIMB_LEFT_LOWER_LEG,
+    /*  4 */ EN_TALK_GIBUD_LIMB_LEFT_FOOT_ROOT,
+    /*  5 */ EN_TALK_GIBUD_LIMB_LEFT_FOOT,
+    /*  6 */ EN_TALK_GIBUD_LIMB_RIGHT_LEG_ROOT,
+    /*  7 */ EN_TALK_GIBUD_LIMB_RIGHT_THIGH,
+    /*  8 */ EN_TALK_GIBUD_LIMB_RIGHT_LOWER_LEG,
+    /*  9 */ EN_TALK_GIBUD_LIMB_RIGHT_FOOT_ROOT,
+    /* 10 */ EN_TALK_GIBUD_LIMB_RIGHT_FOOT,
+    /* 11 */ EN_TALK_GIBUD_LIMB_UPPER_BODY_ROOT,
+    /* 12 */ EN_TALK_GIBUD_LIMB_UPPER_BODY, // Root of Torso
+    /* 13 */ EN_TALK_GIBUD_LIMB_TORSO,      // Root of Left Arm Root, Right Arm Root, and Head Root
+    /* 14 */ EN_TALK_GIBUD_LIMB_LEFT_ARM_ROOT,
+    /* 15 */ EN_TALK_GIBUD_LIMB_LEFT_SHOULDER_AND_UPPER_ARM,
+    /* 16 */ EN_TALK_GIBUD_LIMB_LEFT_FOREARM,
+    /* 17 */ EN_TALK_GIBUD_LIMB_LEFT_HAND,
+    /* 18 */ EN_TALK_GIBUD_LIMB_RIGHT_ARM_ROOT,
+    /* 19 */ EN_TALK_GIBUD_LIMB_RIGHT_SHOULDER_AND_UPPER_ARM,
+    /* 20 */ EN_TALK_GIBUD_LIMB_RIGHT_FOREARM,
+    /* 21 */ EN_TALK_GIBUD_LIMB_RIGHT_HAND,
+    /* 22 */ EN_TALK_GIBUD_LIMB_HEAD_ROOT,
+    /* 23 */ EN_TALK_GIBUD_LIMB_HEAD,
+    /* 24 */ EN_TALK_GIBUD_LIMB_PELVIS,
+    /* 25 */ EN_TALK_GIBUD_LIMB_MAX,
+} EnTalkGibudLimbs;
+
 struct EnTalkGibud;
 
 typedef void (*EnTalkGibudActionFunc)(struct EnTalkGibud*, GlobalContext*);
 
 typedef struct EnTalkGibud {
-    /* 0x0000 */ Actor actor;
-    /* 0x0144 */ char unk_144[0x90];
-    /* 0x01D4 */ EnTalkGibudActionFunc actionFunc;
-    /* 0x01D8 */ char unk_1D8[0x220];
+    /* 0x000 */ Actor actor;
+    /* 0x144 */ ColliderCylinder collider;
+    /* 0x190 */ SkelAnime skelAnime;
+    /* 0x1D4 */ EnTalkGibudActionFunc actionFunc;
+    /* 0x1D8 */ Vec3f limbPos[15];
+    /* 0x28C */ s32 limbIndex;
+    /* 0x290 */ s32 requestedItemIndex;
+    /* 0x294 */ s32 itemActionParam;
+    /* 0x298 */ s32 switchFlag;
+    /* 0x29C */ f32 effectAlpha;
+    /* 0x2A0 */ f32 effectScale;
+    /* 0x2A4 */ Vec3s jointTable[26];
+    /* 0x340 */ Vec3s morphTable[26];
```

Comment:
Hmm, there's something funny going on wth the limbcount, then... is the enum correct?

Reply:
I think its missing `LIMB_NONE`.

Reply:
According to z64utils, yeah, that seems to be it:
Gibdo:
![image](https://user-images.githubusercontent.com/12245827/144696591-732bf7fb-0fbc-4755-90c8-63f67bbd6d60.png)

Redead:
![image](https://user-images.githubusercontent.com/12245827/144696594-11d0f6e1-913f-4620-af4d-6f27cb13bbc4.png)

But it could be wrong. and I have really no way of knowing.

Reply:
Yeah, you need to add one on: the limbs in the object do not include the first one the SkelAnime system uses, which is the position, i.e. `LIMB_NONE`. So unfortunately your enum is going to be off.

Once you've fixed that, you can use the `MAX` entry in the actor struct and the SkelAnime_Init.

Reply:
Yeah, `PostLimbDraw` makes *way* more sense with `LIMB_NONE` being 0, so this has to be it. Thanks for catching this; it's my first time documenting limbs like this.